DataConnectors/CEF/cef_troubleshoot.py

```diff
@@ -110,38 +110,41 @@ def check_red_hat_firewall_issue():
         if "running" in str(o):
             print_warning(
                 "Warning: you have a firewall running on your linux machine this can prevent communication between the syslog daemon and the omsagent.")
-            print("Checking if firewall has exception for omsagent port.[" + agent_port + "]")
-            if red_hat_firewall_d_exception_for_omsagent():
+            print_notice("Checking if firewall has exception for omsagent port [" + agent_port + "]")
+            if validate_rh_firewall_exception():
                 print_ok("Found exception in the firewalld for the omsagent port.[" + agent_port + "]")
-                restart_red_hat_firewall_d()
             else:
                 print_warning("Warning: no exception found for omsagent in the firewall")
                 print_warning(
                     "You can add exception for the agent port[" + agent_port + "] by using the following commands:")
                 print_warning("Add exception:")
-                print_notice(
-                    "sudo firewall-cmd --direct --add-rule ipv4 filter INPUT 0 -p tcp --dport " + agent_port + "  -j ACCEPT")
-                print_warning("Validate the exception was added in the configuration:")
-                print_notice("sudo firewall-cmd --direct --get-rules ipv4 filter INPUT")
+                print_notice("sudo firewall-cmd --direct --permanent --add-rule ipv4 filter INPUT 0 -p tcp --dport " + agent_port + "  -j ACCEPT")
                 print_warning("Reload the firewall:")
                 print_notice("sudo firewall-cmd --reload")
+                print_warning("Validate the exception was added in the configuration:")
+                print_notice("sudo firewall-cmd --direct --get-rules ipv4 filter INPUT")
                 print_warning("You can disable your firewall by using this command - not recommended:")
                 print_notice("sudo systemctl stop firewalld")
 
 
-def red_hat_firewall_d_exception_for_omsagent():
-    '''
-    Check that the firewall_d has an exception for the omsagent
-    :return:
-    '''
-    print("Checking for exception for omsagent")
-    print_notice(firewall_d_exception_configuration_file)
-    firewall_status = subprocess.Popen(["sudo", "cat", firewall_d_exception_configuration_file], stdout=subprocess.PIPE)
-    o, e = firewall_status.communicate()
-    if e is not None:
-        print_error("Error: could not get /etc/firewalld/zones/public.xml file holding firewall exceptions")
-    print_command_response(str(o))
-    return agent_port in str(o)
+def validate_rh_firewall_exception():

```

Comment:
Can you add some docs about what the function is doing and how?

Reply:
Done.